[irrlicht] CMakeLists port file does not read environment variable correctly (patch provided)

**Host Environment**
- OS: Windows 10
- Compiler: VS 2019

**To Reproduce**
Steps to reproduce the behavior:
```
set DXSDK_DIR=C:\Program Files (x86)\Microsoft DirectX SDK (June 2010)
set VCPKG_KEEP_ENV_VARS=DXSDK_DIR
vcpkg.exe install irrlicht --recurse
```
**Failure logs** 
buildtrees\irrlicht\install-x64-windows-v142-dbg-out.log:

```
...\buildtrees\irrlicht\src\1.8.4-e1f2402469\source\Irrlicht\CD3D9ShaderMaterialRenderer.h(15): fatal error C1083: Cannot open include file: 'd3dx9shader.h': No such file or directory
[32/233] C:\PROGRA~2\MIB055~1\2019\PROFES~1\VC\Tools\MSVC\1424~1.283\bin\Hostx64\x64\cl.exe   /TP -DBZ_IMPORT -DIRRLICHT_EXPORTS -DIrrlicht_EXPORTS -DNO_IRR_USE_NON_SYSTEM_BZLIB_ -DNO_IRR_USE_NON_SYSTEM_JPEG_LIB_ -DNO_IRR_USE_NON_SYSTEM_LIB_PNG_ -DNO_IRR_USE_NON_SYSTEM_ZLIB_ -D_CRT_SECURE_NO_WARNINGS -I...\buildtrees\irrlicht\src\1.8.4-e1f2402469\source\Irrlicht -I...\buildtrees\irrlicht\src\1.8.4-e1f2402469\source\Irrlicht\aesGladman -I...\installed\x64-windows-v142\include -I...\buildtrees\irrlicht\src\1.8.4-e1f2402469\source\Irrlicht\lzma -I...\buildtrees\irrlicht\src\1.8.4-e1f2402469\include -I...\buildtrees\irrlicht\src\1.8.4-e1f2402469\ENV{DXSDK_DIR}\Include /DWIN32 /D_WINDOWS /W3 /utf-8 /GR /EHsc /MP  /D_DEBUG /MDd /Z7 /Ob0 /Od /RTC1    /MP /showIncludes /FoCMakeFiles\Irrlicht.dir\source\Irrlicht\CD3D9NormalMapRenderer.cpp.obj /FdCMakeFiles\Irrlicht.dir\ /FS -c ...buildtrees\irrlicht\src\1.8.4-e1f2402469\source\Irrlicht\CD3D9NormalMapRenderer.cpp
FAILED: CMakeFiles/Irrlicht.dir/source/Irrlicht/CD3D9NormalMapRenderer.cpp.obj 
```
Relevant part:
```-I...\buildtrees\irrlicht\src\1.8.4-e1f2402469\ENV{DXSDK_DIR}\Include```

**Additional context**
Patch:
```
diff --git a/ports/irrlicht/CMakeLists.txt b/ports/irrlicht/CMakeLists.txt
index e17d718d8..d7e0c8f50 100644
--- a/ports/irrlicht/CMakeLists.txt
+++ b/ports/irrlicht/CMakeLists.txt
@@ -16,7 +16,7 @@ set(IRR_TOOL_DIR "tools")
 # -------------------------------------------------------------------------------------------------
 set(DXSDK "")
 if(DEFINED ENV{DXSDK_DIR})
-    set(DXSDK "ENV{DXSDK_DIR}")
+    set(DXSDK "$ENV{DXSDK_DIR}")
 endif()
 
 set(IRR_UNICODE_PATH 0 CACHE BOOL "Whether to enable unicode path support on windows")
```

--- a/ports/irrlicht/CMakeLists.txt
+++ b/ports/irrlicht/CMakeLists.txt
@@ -16,7 +16,7 @@ set(IRR_TOOL_DIR "tools")
 # -------------------------------------------------------------------------------------------------
 set(DXSDK "")
 if(DEFINED ENV{DXSDK_DIR})
-    set(DXSDK "ENV{DXSDK_DIR}")
+    set(DXSDK "$ENV{DXSDK_DIR}")
 endif()
 
 set(IRR_UNICODE_PATH 0 CACHE BOOL "Whether to enable unicode path support on windows")

--- a/ports/irrlicht/CONTROL
+++ b/ports/irrlicht/CONTROL
@@ -3,6 +3,7 @@ Version: 1.8.4-5
 Homepage: http://irrlicht.sourceforge.net
 Description: Irrlicht lightning fast 3d engine
 Build-Depends: zlib, libpng, bzip2, libjpeg-turbo
+Supports: !(arm|uwp|osx)
 
 Feature: unicode
 Description: Support unicode path on windows

--- a/ports/irrlicht/portfile.cmake
+++ b/ports/irrlicht/portfile.cmake
@@ -1,3 +1,5 @@
+vcpkg_fail_port_install(ON_ARCH "arm" ON_TARGET "osx" "uwp")
+
 vcpkg_download_distfile(ARCHIVE
     URLS "https://downloads.sourceforge.net/project/irrlicht/Irrlicht%20SDK/1.8/1.8.4/irrlicht-1.8.4.zip"
     FILENAME "irrlicht-1.8.4.zip"
@@ -12,9 +14,7 @@ vcpkg_extract_source_archive_ex(
         "fix-encoding.patch"
 )
 
-# Copy CMakeLists.txt to the source, because Irrlicht does not have one.
-file(COPY ${CMAKE_CURRENT_LIST_DIR}/CMakeLists.txt DESTINATION ${SOURCE_PATH})
-file(COPY ${CMAKE_CURRENT_LIST_DIR}/LICENSE.txt DESTINATION ${SOURCE_PATH})
+configure_file(${CMAKE_CURRENT_LIST_DIR}/CMakeLists.txt ${SOURCE_PATH}/CMakeLists.txt COPYONLY)
 
 vcpkg_check_features(
     OUT_FEATURE_OPTIONS FEATURE_OPTIONS
@@ -24,10 +24,7 @@ vcpkg_check_features(
         tools       IRR_BUILD_TOOLS
 )
 
-set(SHARED_LIB TRUE)
-if(VCPKG_LIBRARY_LINKAGE STREQUAL "static")
-    set(SHARED_LIB FALSE)
-endif()
+string(COMPARE EQUAL "${VCPKG_LIBRARY_LINKAGE}" "dynamic" SHARED_LIB)
 
 vcpkg_configure_cmake(
     SOURCE_PATH ${SOURCE_PATH}
@@ -52,7 +49,5 @@ vcpkg_copy_pdbs()
 if(VCPKG_LIBRARY_LINKAGE STREQUAL "static")
     file(COPY ${CMAKE_CURRENT_LIST_DIR}/vcpkg-cmake-wrapper.cmake DESTINATION ${CURRENT_PACKAGES_DIR}/share/irrlicht)
 endif()
-# Post-build test for cmake libraries
-vcpkg_test_cmake(PACKAGE_NAME irrlicht)
 
-file(INSTALL ${SOURCE_PATH}/LICENSE.txt DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT} RENAME copyright)
+file(INSTALL ${CMAKE_CURRENT_LIST_DIR}/LICENSE.txt DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT} RENAME copyright)

